specs/RestartRequested.md

```diff
@@ -20,11 +20,20 @@ void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2Init
     }
 }
 
-void WebView_RestartRequested(CoreWebView2Environment sender, object e)
+void WebView_RestartRequested(CoreWebView2Environment sender, CoreWebView2RestartRequestedEventArgs e)
 {
-    // Depending on your app experience, you may or may not
-    // want to prompt user to restart the app.
-    RestartIfSelectedByUser();
+    if (e.Priority == RestartRequestedPriority.Normal) 
```

Comment:
Not sure how much you want to future-proof. This sample is assuming it will stay a two state enum. You could make this a <= check (maybe change Normal to be above zero), and leave room to have multiple forms of high pri.

Alternatively, if it's really going to stay a two-state, then make it a bool to avoid confusion.

Reply:
Windows OS and WASDK have examples where Lower-than-normal priority enums have negative values, but I don't know if that's appropriate for WV2.

(e.g. https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.dispatching.dispatcherqueuepriority?view=windows-app-sdk-1.5 )

Using an "IsHighPriotity" bool now still allows us to add a 3+ level enum property later if needed, but we'd have to partition them into "High=true" and "High=false" buckets at that time.

Reply:
I think either bool or <= check is OK.  I'm slightly inclined toward <=, and use a positive number for .Normal if we don't support negative values.

I'm inclined to define an enum now instead of using a bool, because I can already think of two possible values we'd add later:  "Low" and a higher-than-"High" value if we wanted to be able to recommend restarting without even waiting for the user prompt mentioned in this sample.

Reply:
Or we can leave gaps in the enum so we can slot new values later.

```
typedef enum COREWEBVIEW2_RESTART_REQUESTED_PRIORITY {
  /// Developer should remind user to restart.
  COREWEBVIEW2_RESTART_REQUESTED_PRIORITY_NORMAL = 1000,
  /// Developer should prompt user to restart as soon as possible. 
  COREWEBVIEW2_RESTART_REQUESTED_PRIORITY_HIGH = 2000,
} COREWEBVIEW2_RESTART_REQUESTED_PRIORITY;
```

Then we can later add `_LOW` as 500 or `_CRITICAL` as 3000.

Maintaining the numeric sequencing allows the app to use greater-than/less-than testing.

Reply:
Frank's [DispatcherQueuePriority](https://docs.microsoft.com/uwp/api/Windows.System.DispatcherQueuePriority) example actually uses both negative numbers and VB spacing :).

Both sound good to me. I wasn't sure if negative numbers would work, because enums can technically be signed or unsigned, but we have a working example. Maybe in the WinRT projection it's always signed.

Reply:
Each language decides whether enums are signed. In C# they are always signed. In JavaScript they are always signed. In C++ they can be either (C++/WinRT projects enums as signed, except that flags enums project as unsigned). I wouldn't be surprised if there's some language that doesn't support negative enums. (It looks like Rust is one of those, but rust-rt works around it by just projecting enums as integer constants!)

Even if signed enums are allowed (or worked around) in all languages, I wouldn't put it past people to get the cast wrong and accidentally cast to unsigned instead of signed.

Reply:
I forgot about [flags], they're unsigned in the c# projection too.

Reply:
Sounds good. Talking with folks there is a possibility of future enum values. We can go with the suggested values and have plenty of room in the future and switch the sample code to use `<= Normal`

```
typedef enum COREWEBVIEW2_RESTART_REQUESTED_PRIORITY {
  /// Developer should remind user to restart.
  COREWEBVIEW2_RESTART_REQUESTED_PRIORITY_NORMAL = 1000,
  /// Developer should prompt user to restart as soon as possible. 
  COREWEBVIEW2_RESTART_REQUESTED_PRIORITY_HIGH = 2000,
} COREWEBVIEW2_RESTART_REQUESTED_PRIORITY;
```

Reply:
Updated.